project_task_pull_request_state/models/project_task.py

```diff
@@ -0,0 +1,47 @@
+from odoo import fields, models, api
+
+
+class ProjectTaskState(models.Model):
+    _inherit = "project.task"
+
+    pr_state = fields.Selection(
+        selection=lambda self: self._selection_pr_state(),
+        string="PR State",
+        )
+
+    @api.model
+    def _selection_pr_state(self):
+        """Function to select the state of the pull request"""
+        return [
+            ('open', 'Open'),
+            ('draft', 'Draft'),
+            ('merged', 'Merged'),
+            ('closed', 'Closed'),
+        ]
+
+    @api.model
+    def create(self, vals):
+        if 'project_id' in vals:
+            if 'pr_uri' in vals:
+                search_pr_state = self.env['project.project'].search(
+                    [('id', '=', vals.get('project_id'))]).pr_state_default
+                vals['pr_state'] = search_pr_state
+            else:
+                vals['pr_state'] = None
+        else:
+            vals['pr_state'] = None
+        task = super(ProjectTaskState, self).create(vals)
+        return task
+
+    def write(self, vals):
+        self._check_pr_state(vals)
+        result = super(ProjectTaskState, self).write(vals)
+        return result
+
+    def _check_pr_state(self, vals):
```

Comment:
This function makes pr_state values updated, but the function name is _check. Why?
Maybe the better solution will be renamed function and return new `pr_state` value?